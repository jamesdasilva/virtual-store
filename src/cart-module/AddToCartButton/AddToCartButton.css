.add-to-cart-button {
  box-sizing: border-box;
  height: 35px;
  border-radius: 17.5px;
  margin: 5px 10px 10px;
  padding: 0;
  border: 0;
  color: #fff;
  background-color: #78BE20;
  cursor: pointer;
}