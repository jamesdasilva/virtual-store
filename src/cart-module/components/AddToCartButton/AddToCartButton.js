import React from 'react'
import PropTypes from 'prop-types'
import './AddToCartButton.css'

const AddToCartButton = ({ addItemToCart, item }) => {
  return (
    <button
      data-testid='add-to-cart-button'
      className='add-to-cart-button'
      onClick={() => addItemToCart(item)}
    >Adicionar ao carrinho</button>
  )
}

AddToCartButton.propTypes = {
  addItemToCart: PropTypes.func,
  item: PropTypes.object
}

export default AddToCartButton
