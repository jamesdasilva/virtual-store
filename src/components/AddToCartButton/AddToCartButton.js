import React from 'react'
import PropTypes from 'prop-types'
import './AddToCartButton.css'

const AddToCartButton = ({ clickHandler, product }) => {
  return (
    <button className='add-to-cart-button' onClick={() => clickHandler(product)}>
      Adicionar ao carrinho
    </button>
  )
}

AddToCartButton.propTypes = {
  clickHandler: PropTypes.func,
  product: PropTypes.object
}

export default AddToCartButton
