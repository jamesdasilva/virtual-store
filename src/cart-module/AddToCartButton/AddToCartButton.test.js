import '@testing-library/jest-dom'
import { render, cleanup, waitForElement, fireEvent } from '@testing-library/react'
import React from 'react'
import AddToCartButton from './AddToCartButton'

afterEach(cleanup)

describe('<AddToCartButton /> spec', () => {
  test('Snapshot test', async () => {
    const addItemToCartSpy = jest.fn()
    const objectMock = { test: "test" }
    const { getByTestId } = render(
      <AddToCartButton
        addItemToCart={addItemToCartSpy}
        item={objectMock} />
      )
    const addToCartButton = await waitForElement(() => getByTestId('add-to-cart-button'))
    expect(addToCartButton).toMatchSnapshot()
  })
  
  test('Should call clickHandler function, passing the item parameter, when clicked', async () => {
    const addItemToCartSpy = jest.fn()
    const objectMock = { test: "test" }
    const { getByTestId } = render(
      <AddToCartButton
        addItemToCart={addItemToCartSpy}
        item={objectMock} />
      )
    const addToCartButton = await waitForElement(() => getByTestId('add-to-cart-button'))
    fireEvent.click(addToCartButton)
    expect(addItemToCartSpy).toBeCalledWith(objectMock)
  })
})
