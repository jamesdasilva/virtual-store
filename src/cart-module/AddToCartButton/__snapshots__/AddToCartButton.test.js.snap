// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddToCartButton /> spec Snapshot test 1`] = `
<button
  class="add-to-cart-button"
  data-testid="add-to-cart-button"
>
  Adicionar ao carrinho
</button>
`;
